text/0000-thread-safe-handles.md

```diff
@@ -0,0 +1,362 @@
+- Feature Name: Threadsafe Handles
+- Start Date: 2020-07-21
+- RFC PR: (leave this empty)
+- Neon Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Two new `Send` and `Sync` features are introduced: `EventQueue: Send + Sync` and `Persistent: Send`. These features work together to allow multi-threaded modules to interact with the JavaScript main thread.
+
+* `EventQueue`: Threadsafe handle that allows sending closures to be executed on the main thread.
```

Comment:
I'd avoid the word "handle" here to avoid confusion with `Handle`. And maybe be explicit that we're talking about Rust closures.

Also, the RFC talks about "main thread" a lot but it's pretty future-proof to work with multiple JS threads once N-API adds support for workers and postMessage. Seems like we can generally say "JavaScript thread" or "the JavaScript thread the value belongs to" etc instead of "main thread."